include/zephyr/linker/common-rom/common-rom-kernel-devices.ld

```diff
@@ -45,6 +45,14 @@
 	} GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 #endif
 
+#if defined(CONFIG_GEN_IRQ_VECTOR_TABLE)
+	SECTION_PROLOGUE(irq_vector_table,,)
+	{
+		. = ALIGN(CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN);
+		*(_IRQ_VECTOR_TABLE_SECTION_SYMS)
```

Comment:
But `_IRQ_VECTOR_TABLE_SECTION_SYMS` are already given as input sections at other places?

https://github.com/zephyrproject-rtos/zephyr/blob/21d1ad3762302b3e461953df59430c77e0709274/arch/arc/core/vector_table.ld#L11

https://github.com/zephyrproject-rtos/zephyr/blob/301243ed06ac77da6810c20898fe0059b4835d61/arch/arm/core/aarch32/vector_table.ld#L44

https://github.com/zephyrproject-rtos/zephyr/blob/806ee8601ca73b0de0767b2a22c533e307bcee8b/include/zephyr/arch/arm64/scripts/linker.ld#L163

Reply:
Oh, interesting. Since I was working on RISCV I missed that.

The problem is that this is half-baked now because without this patch each arch must adapt its own linker script file to place that and that's inconvenient at least.

Reply:
@stephanosio what about moving this to a standalone `.ld` file to be included by the arch? So we can keep supporting also `CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN`.

Reply:
> what about moving this to a standalone .ld file to be included by the arch?

That sounds reasonable and is probably what should have been done in the first place when this feature was added.